[new-product] InfluxDB

bentasker opened the pull request:
Add [InfluxDB](https://github.com/influxdata/influxdb) (2.7.x and 3.0.x are the currently supported versions)

Note: looks like this would also close https://github.com/endoflife-date/endoflife.date/issues/6178

bentasker committed:
docs: add InfluxDB

Signed-off-by: Ben Tasker <2900301+bentasker@users.noreply.github.com>

captn3m0 committed:
Update influxdb.md

marcwrobel commented on products/influxdb.md:
```diff
@@ -0,0 +1,45 @@
+category: server-app
+iconSlug: influxdb
+permalink: /influxdb
+versionCommand: influxd version
```
```suggestion
versionCommand: influx version
```

Based on https://docs.influxdata.com/influxdb/cloud/tools/influx-cli/.

bentasker replied:
`influx version` gives you the version of the v2 CLI rather than of InfluxDB itself - you need to run `influxd version` on the host itself

marcwrobel committed:
Update releases

marcwrobel commented on products/influxdb.md:
```diff
@@ -0,0 +1,60 @@
+  methods:
+  -   git: https://github.com/influxdata/influxdb
+
+# eol(x) = releaseDate(x+1)
```
```suggestion
# eol(x) = releaseDate(x+2)
```

Based on https://www.influxdata.com/legal/support-policy/:

>  InfluxData will provide Support Services for the current Minor Release (x.Y) of the Software and the immediately preceding Minor Release (x.Y) of the Software...

What about eoas ? I did not find any mention of a different support level, so maybe eoas should be dropped ?

bentasker replied:
Good call - I'll remove it

marcwrobel commented on products/influxdb.md:
```diff
@@ -0,0 +1,60 @@
+
+---
+
+> [InfluxDB](https://github.com/influxdata/influxdb) is an open source time series platform.
```
Is this about open source or commercial ? Because the release policy link (https://www.influxdata.com/legal/support-policy/) only applies for users with a commercial licence.

bentasker replied:
It's... complicated... :) 

Technically, the support policy *is* only for users with a commercial license.

However: 

- There has never been a commercial release of on-prem 2.x (commercial 2.x is/was SaaS only)
- The [Open Source maintenance policy](https://docs.influxdata.com/influxdb/v2/reference/release-notes/supported-release/) for 2.x specifically links to that doc as a reference

marcwrobel replied:
Looking at https://docs.influxdata.com/influxdb3/core/release-notes/ it seems that since 3 the OSS version is now named "Core". This seem confirmed by https://itbrief.news/story/influxdb-3-core-enterprise-bring-high-speed-data-tools.

If this is correct I think we should document this clearly and change the name of the page so that there is no ambiguity.

And based on those information we could say:

- InfluxDB follows an open-core model
- there are no documented support policy for InfluxDB Core
- but InfluxDB Core follows the same release policy than Enterprise (same release notes, such as https://docs.influxdata.com/influxdb3/core/release-notes/),
- so some information from https://www.influxdata.com/legal/support-policy/ should be true for both Core and Enterprise.

WDYT ?

bentasker replied:
Yep, all makes sense to me - have just pushed changes to incorporate this.

marcwrobel commented on products/influxdb.md:
```diff
@@ -0,0 +1,60 @@
+
+-   releaseCycle: "2.7"
+    releaseDate: 2023-04-05
+    eol: false # 3.1 release
```
Should be 12 months according to https://www.influxdata.com/legal/support-policy/ :

> In the event InfluxData releases a Major Release (X.y), InfluxData will provide Support Services for the current Major Release (X.y) of the Software and the immediately preceding Minor Release (x.Y) of the Software, inclusive of all Maintenance Releases (x.y.Z) included in the respective Minor Release for at least 12 months from the InfluxData general availability of the Major Release. (By way of example only, InfluxData releases 2.0 (Major release version 2, Minor release version 0) and several maintenance releases thereafter (2.0.1, 2.0.2 and so forth). In this example, InfluxData would provide Support Services not only for 2.0 and its maintenance releases, but also 1.9 and all of its respective maintenance releases for at least 12 months from the InfluxData general availability of 2.0).

bentasker replied:
Good catch, updated the comment

marcwrobel commented on products/influxdb.md:
```diff
@@ -0,0 +1,60 @@
+---
+title: InfluxDB
+category: server-app
```
```suggestion
category: server-app
tags: database influxdata
```

bentasker committed:
Update products/influxdb.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

bentasker committed:
Update products/influxdb.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

bentasker committed:
fix: correct expected EOL

bentasker committed:
fix: remove end of actual support, there is no documented eoas

bentasker committed:
docs: rename page and incorporate PR feedback

bentasker committed:
docs: adjust wording

marcwrobel commented on products/influxdb-core.md:
```diff
@@ -0,0 +1,64 @@
+> It includes APIs for storing and querying data, processing it in the background for ETL or
+> monitoring and alerting purposes, user dashboards, and visualizing the data.
+
+Starting from the release of InfluxDB OSS 3, the product became known as InfluxDB 3 Core and
```
Just read https://en.wikipedia.org/wiki/InfluxDB#InfluxDB_3_changes_open_source_offerings and it states that InfluxDB core is not a replacement for InfluxDB OSS. Looking at the announcement, https://www.influxdata.com/blog/influxdb-3-oss-ga/, it makes sense.

I am not familiar with InfluxDB, so can't really tell whether this is really different or not. WDYT @bentasker ? Is InfluxDB Core 3 on par with InfluxDB OSS 2 in terms of features ?

bentasker replied:
Yeah, it's a bit of a weird one.

Officially, InfluxDB 3 **is** the _future_, but 3.x is also a very different product in terms of capabilities, features and use-case.

1.x/2.x are full time-series databases with no restriction on how much data you can retain (or how long for). It's not uncommon to see OSS users utilising it for long-term storage (including [writing homeassistant data into it](https://www.home-assistant.io/integrations/influxdb/) etc).


In comparison, 3.x:

* drops support for the Flux query language in favour of SQL
* drops support for flux tasks (though now has the processing engine instead)
* drops the integrated UI
* There are various differences in the tooling, including binary naming (`influxd` vs `influxdb3`)
* completely reimplements the storage engine: it uses datafusion + parquet rather than TSM (although this might _sound_ like an implementation detail, it has an impact on what **good** schema design looks like) 
* Because of the lack of compactor, OSS has a practical limit on the amount of data that you can query in one go
* (OSS) Limits the number of databases to five
* It also introduces a bunch of _new_ features (Last value cache etc)

Is your line of thinking here that we should perhaps break out into two products/pages? One for InfluxDB 1&2x and one for 3.x/Core?

If so, that would make sense to me. I obviously don't **know**, but I suspect that 1.x and 2.x will continue to see releases for longer than the support policy implies - 1.x enterprise still gets active development (and there have been [a couple](https://github.com/influxdata/influxdb/releases/tag/v1.11.0) of [recent bumps](https://github.com/influxdata/influxdb/releases/tag/v1.12.0) in the OSS versions too). 

2.x cloud is also still an ongoing thing, so fixes made there (particularly to common elements like Flux) will likely bubble into 2.x OSS.

marcwrobel replied:
> Is your line of thinking here that we should perhaps break out into two products/pages? One for InfluxDB 1&2x and one for 3.x/Core?

That would indeed makes things clearer. What I would expect is that if you are using 2.x OSS, an upgrade path to 3.x exists and does not involve subscribing to the enterprise version. For InfluxDB 3 I have the impression this is not the case at all.

bentasker replied:
Yep, that's correct.

Influxdata **are** giving away [free enterprise subscriptions to home users](https://www.influxdata.com/blog/influxdb3-open-source-public-alpha-jan-27/) but that still entails running non-OSS components.

There's a question on the forum about that license [here](https://community.influxdata.com/t/questions-about-influxdb-3-enterprise-free-tier-for-home-users-and-the-future-of-influxdb-v2/57028) with the reply also noting

> v2 continues to remain supported with security/bug fix updates, and has no EOL planned currently. We do encourage users to move to InfluxDB 3, however, as we’ll be continuing to develop an extensive feature set on our roadmap.

I'll split the pages out and crosslink them to each other

bentasker replied:
Changes pushed - I've also added the latest 1.x cycle to `influxdb.md`

bentasker replied:
Oh, maybe I need to take that back out, lint doesn't like it

> Product Validator: Invalid releases '1.12' for influxdb.md, expecting release (released on 2025-04-08) to be before 2.7 (released on 2023-04-05).

bentasker replied:
Hey @marcwrobel, just checking in - need anything else from me here?

bentasker committed:
docs: split influxdb and influxdb-core into seperate products

bentasker committed:
fix: correct version command for influxdb-core

bentasker committed:
feat: add the latest 1.x series release

bentasker committed:
fix: remove influxdb 1.x

Lint CI doesn't like the latest minor release of 1.x being more recent than 2.x

marcwrobel committed:
Merge pages

marcwrobel reviewed (commented):
Hi @bentasker, sorry the long time since my last review.

I have been reviewing the 2-products version but was not really happy about it. 

I have just committed a proposition, visible on https://deploy-preview-7303--endoflife-date.netlify.app/influxdb, where I:

- merged back both pages,
- focused on the commercial version (it makes things easier to explain),
- and talked about the OSS versions / the change with InfluxDB 3 in the last paragraphs.

WDYT of this proposition ?

bentasker committed:
docs: add link to release notes for 1.x and 2.x

marcwrobel committed:
addedAt

marcwrobel reviewed (approved):
LGTM, thanks @bentasker, it was a difficult one !
